tutorials/tts/TTS_pronounciation_customization.ipynb

```diff
@@ -0,0 +1,290 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Default G2P\n",
+    "\n",
+    "Below we show how to analyze default G2P output of the NeMo models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import nemo.collections.tts as nemo_tts\n",
+    "from nemo_text_processing.g2p.modules import IPAG2P\n",
+    "import soundfile as sf\n",
+    "import IPython.display as ipd\n",
+    "import torch\n",
+    "\n",
+    "# Load mel spectrogram generator\n",
+    "spec_generator = nemo_tts.models.FastPitchModel.from_pretrained(\"tts_en_fastpitch_ipa\").eval()\n",
+    "# Load vocoder\n",
+    "vocoder = nemo_tts.models.HifiGanModel.from_pretrained(model_name=\"tts_hifigan\").eval()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# to use dictionary entries for known words\n",
+    "spec_generator.vocab.phoneme_probability = 1\n",
+    "spec_generator.vocab.g2p.phoneme_probability = 1\n",
+    "\n",
+    "# helper fucntion to generate audio\n",
+    "def generate_audio(input_text):\n",
+    "    parsed = spec_generator.parse(input_text)\n",
+    "    spectrogram = spec_generator.generate_spectrogram(tokens=parsed)\n",
+    "    audio = vocoder.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "    display(ipd.Audio(audio.detach().to('cpu').numpy(), rate=22050))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "text = \"paracetamol can help reduce fever.\"\n",
+    "generate_audio(text)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "During preprocessing, unambiguous dictionary words are converted to phonemes, while OOV and words with multiple entries are kept as graphemes. For example, **paracetamol** is missing in from the phonetic dictionary, and **can** has 2 forms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Input before tokenization: |{' '.join(spec_generator.vocab.g2p(text))}|\\n\")\n",
+    "for word in [\"paracetamol\", \"can\"]:\n",
+    "    word = word.upper()\n",
+    "    phoneme_forms = spec_generator.vocab.g2p.phoneme_dict[word]\n",
+    "    print(f\"Number of phoneme forms for wordPhoneme forms for '{word}': {len(phoneme_forms)} -- {phoneme_forms}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Input customization\n",
+    "\n",
+    "One a one-time customization suffice, one can pass phonemes directly as input to the model.\n",
+    "Let's replace the word **paracetamol** with the desired phonetic form in our text example."
```

Comment:
Might be helpful to users to explicitly say they need to put `|` around each symbol in the phonetic input.

Reply:
done